File: examples/minimal-cron.rs
use ancymon::{handlers::DebugHandler, triggers::cron::CronTrigger, Bot, Config};
use std::fs;

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt()
        .with_max_level(tracing::Level::DEBUG)
        .init();

    let config_str = fs::read_to_string("examples/minimal-cron.lua").unwrap();
    let config = Config::new(&config_str).unwrap();

    Bot::default()
        .with_handler_type("debug", DebugHandler)
        .with_source_type("cron", CronTrigger::default())
        .run(config)
        .await
        .unwrap();
}

File: examples/minimal-cron.lua
sources = {
	["cron"] = {
		["type"] = "cron",
	},
}

triggers = {
	{
		["source"] = "cron",
		["emit"] = "2-trigger",
		["arguments"] = "*/2 * * * * *",
	},
	{
		["source"] = "cron",
		["emit"] = "3-trigger",
		["arguments"] = "*/3 * * * * *",
	},
}

handlers = {
	["debug"] = {
		["type"] = "debug",
	},
}

actions = {
	{
		["handler"] = "debug",
		["event"] = "2-trigger",
		["emit"] = "2-debug",
	},
	{
		["handler"] = "debug",
		["event"] = "3-trigger",
		["emit"] = "3-debug",
	},
}
